sync_ci/pkg/parser/regex_rules_test.go

```diff
@@ -0,0 +1,84 @@
+package parser
+
+import (
+	"bufio"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestRegex_TestRules(t *testing.T) {
+	updateRegexRules("./pkg/parser/regex_rules.json")
+	lines := FilesToLines("./pkg/parser/rules_test/integration_fatal_error.log")
+	result := ApplyRegexpRulesToLines("tidb_ghpr_check", lines)
+	if result == nil || len(*result) != 2 {
+		t.Error("test failed")
+		return
+	}
+
+	for _, r := range *result {
+		if r.Key != "case" ||
+			r.Value != `[FATAL] [main.go:694] ["run test"] [test=select] [error="sql:SELECT c2 from t where not (c2 > 2);: run \"SELECT c2 from t where not (c2 > 2);\" at line 85 err, we need` {
+			t.Error("test failed")
+		}
+	}
+}
+
+func FilesToLines(path string) []string {
+	file, err := os.Open(path)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	lines := make([]string, 0)
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+
+	return lines
+}
+
+func TestRegexp_TiKVCompileError(t *testing.T) {
+	updateRegexRules("./pkg/parser/regex_rules.json")
+	lines := FilesToLines("./pkg/parser/rules_test/tikv_compile_error.log")
+
+	info := ApplyRegexpRulesToLines("tikv_ghpr_test", lines)
+	for _, o := range *info {
+		if o.Key == "compile" && strings.Contains(o.Value, "error: could not compile") {
+			return
+		}
+	}
+
+	t.Error("TiKV compile error missed.")
+}
+
+func TestRegexp_TiDBPanic(t *testing.T) {
+	updateRegexRules("./pkg/parser/regex_rules.json")
+	lines := FilesToLines("./pkg/parser/rules_test/tidb_panic.log")
+
+	info := ApplyRegexpRulesToLines("tidb_ghpr_check", lines)
+	for _, o := range *info {
+		if o.Key == "case" && strings.Contains(o.Value, "PANIC: aggregate_test.go:507: testSuiteAgg.TestGroupConcatAggr") {
```

Comment:
When the rules are matched, some additional operations are needed to combine the results we expect.I think in this case, it is not suitable to configure rules in the file.

Reply:
It is so bad that golang regexp does not support lookaround, which could do the job you want. But filtering out cases, including some other information is ok to me.

Reply:
Additional information such as number `507` of line may affect our judgment on the uniqueness of the use case.

Reply:
As we talk earlier, we can take this as part of this failed case. That will not matter so much.